packages/teams-js/test/resultValidation.ts

```diff
@@ -0,0 +1,52 @@
+import { MessageRequest } from './utils';
+
+export enum MatcherType {
+  ToBe,
+  ToStrictEqual,
+}
+
+// This function will return the number of assertions used to validate a request object using
+// validateRequestWithoutArguments or validateExpectedArgumentsInRequest.
+// The number of assertions used changes depending on how many arguments are being validated, so
+// you have to pass in the number of arguments being validated.
+// The value returned from this function can be used in calls to expect.assertions() if you have them.
+export function getNumberOfAssertionsUsedToValidateRequest(numberOfArgumentsBeingValidated: number): number {
+  return 2 + (numberOfArgumentsBeingValidated === 0 ? 0 : 2 + numberOfArgumentsBeingValidated);
+}
+
+// Used to validate a request object you are expecting to contain no arguments.
+export function validateRequestWithoutArguments(request: MessageRequest | null, expectedFunctionName: string): void {
+  validateExpectedArgumentsInRequest(request, expectedFunctionName, MatcherType.ToBe);
+}
+
+/* The following two lint rules are disabled for only this function since this function is specifically testing
+ * for null and undefined as part of validation and then using those values after testing them. */
+/* eslint-disable strict-null-checks/all */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+// Used to validate a request object you are expecting to contain arguments.
+export function validateExpectedArgumentsInRequest(
+  request: MessageRequest | null,
+  expectedFunctionName: string,
+  matcher: MatcherType,
+  ...expectedArgs: unknown[]
+): void {
+  expect(request).not.toBeNull();
+  expect(request?.func).toEqual(expectedFunctionName);
+  if (expectedArgs.length > 0) {
+    expect(request!.args).toBeDefined();
+    expect(request!.args!.length).toBe(expectedArgs.length);
+
+    for (let i = 0; i < expectedArgs.length; ++i) {
```

Comment:
I would argue that `++i` is the usual, but I guess it depends on what circles you run in.  😁

If you don't need the behavior of post-increment, you should always use pre-increment because it _can_ be more efficient. This can certainly vary across languages, technologies, runtimes, etc. However, AFAIK it's not ever the case that pre-increment is less efficient, it just might be as efficient. Even if efficiency were not a factor, I would just go back to "if you aren't depending on the side-effects of post-increment then don't use it."

https://stackoverflow.com/questions/24901/is-there-a-performance-difference-between-i-and-i-in-c